### DISCUSSION TITLE
Add new chapters to M4B files?

### DISCUSSION THREAD

**Discussion by nhan000** [yermak/AudioBookConverter#504] [Q&A]
I'd like to add new chapters and edit existing chapters' names and timestamps in my M4B files, preferably without re-encoding. Is there a way to do that? I see that there's an issue here https://github.com/yermak/AudioBookConverter/issues/451 but there's no reply.

**Comment by yermak** [comment]
Depending what do you mean by edit chapters/timestamps... I believe program allows this.
Open m4b file -> It will appear as single file
Press "Chapters" -> it will load existing chapters into view.
Select a chapter, and right-click on it. Choose Edit -> You could change name manually or use existing meta information.
You could select 2 and more chapters and merge them, then you could split them by selecting track where you want to split
You could select track under chapter and split it but sub-tracks (currently only fixed duration is supported, no exact timestamp)

**Reply by nhan000** [reply]
Thanks for the answer! Does it also allows me to *add new chapters* as well or only edit existing chapters?

**Reply by yermak** [reply]
You could add more chapters either from mp3 file or from m4b.

You could even mix and match chapters from another book.

Add m4b book and load chapters, add another m4b book (for second book chapters will load automatically as you are in chapters mode). Alternatively you could add 2 or more m4b books and then load chapters.
After this you could move chapters up and down, delete some of them or merge multiple chapters into single one or split existing chapter into multiple tracks.

If you have specific use case please feel to share - I could advise solution to that.

**Reply by HyperEpsilon** [reply]
What if my audio file does not have any existing chapters? I have a .ogg file, but I want to add chapters to it, how would one do that?

**Reply by nhan000** [reply]
Hi @yermak, @HyperEpsilon's question was what I meant to ask: to add new chapters to an m4b file that doesn't have any existing chapters, or add new chapters to a file that already has some chapters. What you described was way more advanced that what I needed.

**Comment by jjones3905** [comment]
@nhan000 It's a little putzy, but you can do it like this:

1) Go to the chapter that will precede the new chapter.
2) Figure out how many seconds into it you want to start the new chapter.
3) Open the chapter to show the underlying track. Note that if you've done a previous sub-tracks call or if it's an m4b, this isn't a file name, just the related chapter name. But the point is, use the twisty by the chapter to open up the details below.
4) Select that and then call up the Sub-tracks menu (button or context menu).
5) Specify the number of seconds you want to remain with the existing chapter. !!! MAKE SURE YOU TURN OFF "Multiple Equal Segments" !!! This means that it's only going to do one split with the first part the given number of seconds long and the second with the remainder. If you fail to turn it off, ABC will happily go making sub-tracks as many times as it can with the duration. So, if you have what currently looks like a 10 minute chapter and you want to insert a new chapter with the old chapter taking up 4 minutes and the new chapter 6 minutes, you'd set the sub-track length to 240, unselect the multiple, leave the chapter creation on and accept. This will do what you want. If you forget to turn the multiple slider off, instead you will get two new chapters of length 4 and 2 minutes respectively.
6) Repeat as needed.

Note that this does not insert some other file than the original m4b. It just changes the chapters specification to the underlying ffmpeg command.

**Reply by jjones3905** [reply]
(If you use that Chapters button, you will have at least one chapter to start with, whether or not your source file has chapters. And, of course, you can only do this for output formats that support chaptering. Well, you can do it, but the output file is going to ignore it all.)
